src/test/java/uk/gov/pay/directdebit/webhook/gocardless/services/GoCardlessEventHandlerForMandatesTest.java

```diff
@@ -48,43 +56,52 @@
             .withPayerFixture(payerFixture);
 
     private GoCardlessEventFixture goCardlessEventFixture = GoCardlessEventFixture.aGoCardlessEventFixture()
+            .withResourceType(GoCardlessResourceType.MANDATES)
             .withLinksOrganisation(organisationIdentifier);
 
     @Test
     public void handle_onAFailedMandateGoCardlessEvent_shouldRegisterAPayEventAsMandateFailed() {
-        GoCardlessEvent goCardlessEvent = spy(goCardlessEventFixture.withAction("failed").toEntity());
+        GoCardlessEvent goCardlessEvent = goCardlessEventFixture.withAction("failed").toEntity();
 
         when(mockMandateQueryService.findByGoCardlessMandateIdAndOrganisationId(
                 goCardlessEvent.getLinksMandate().get(), goCardlessEventFixture.getLinksOrganisation()))
                 .thenReturn(mandateFixture.toEntity());
 
-        goCardlessMandateHandler.handle(goCardlessEvent);
+        goCardlessEventHandler.handle(List.of(goCardlessEvent));
 
         verify(userNotificationService).sendMandateFailedEmailFor(mandateFixture.toEntity());
+        verify(mockMandateQueryService).findByGoCardlessMandateIdAndOrganisationId(
+                goCardlessEvent.getLinksMandate().get(), goCardlessEventFixture.getLinksOrganisation());
+        verify(mockPaymentQueryService, never()).findByGoCardlessPaymentIdAndOrganisationId(any(), any());
```

Comment:
I don’t think this is necessary, there’s no real harm if the code does a pointless lookup. Might be worth verifying there’s no more interactions with the `userNotificationService` though.

Reply:
fixed